File: app/src/main/java/ru/nikshlykov/englishwordsapp/di/ContextModule.java
package ru.nikshlykov.englishwordsapp.di;

import android.app.Application;
import android.content.Context;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
public class ContextModule {
    @Provides
    Context provideContext(Application application){
        return application;
    }
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/di/ContextModule.kt
package ru.nikshlykov.englishwordsapp.di

import android.app.Application
import android.content.Context
import android.speech.tts.TextToSpeech
import dagger.Module
import dagger.Provides
import ru.nikshlykov.englishwordsapp.App
import javax.inject.Singleton

@Module
class ContextModule {
  @Provides
  fun provideContext(application: Application): Context {
    return application
  }

  @Provides
  @Singleton
  fun provideTextToSpeech(application: Application): TextToSpeech {
    return (application as App).textToSpeech!!
  }
}